File: java/pessoal.py
from compromisso import Compromisso

class Pessoal(Compromisso):
    def __init__(self, id, data, hora, duracao, prioridade, adiavel, motivo, local):
        super().__init__(id, 'P', data, hora, duracao, prioridade, motivo)
        self.adiavel = adiavel
        self.local = local
        self.calcularPrioridadeFinal()
    
    def calcularPrioridadeFinal(self):
        multiplicador = 2
        if not self.adiavel:
            multiplicador += 1
        self.prioridadeFinal = self.prioridade * multiplicador
    
    def getDescricao(self):
        return self.descricao
    
    def isAdiavel(self):
        return self.adiavel
    
    def getLocal(self):
        return self.local

File: java/Pessoal.java
public class Pessoal extends Compromisso{
    private boolean adiavel;
    private String local;

    public Pessoal(int id, Data data, Hora hora, int duracao, int prioridade, boolean adiavel, String motivo, String local){
        super(id, 'P', data, hora, duracao, prioridade, motivo);
        this.adiavel = adiavel;
        this.local = local;
        calcularPrioridadeFinal();
    }

    @Override
    public void calcularPrioridadeFinal(){
        //multiplicador 2x
        int multiplicador = 2;
        //se nao adiável +1x
        if(!adiavel){ 
            multiplicador += 1;
        }
        this.prioridadeFinal = this.prioridade * multiplicador;
    }

    @Override
    public String getDescricao(){
        return this.descricao;
    }

    @Override
    public boolean isAdiavel(){
        return adiavel;
    }

    public String getLocal(){
        return local;
    }
}